esrally/telemetry.py

```diff
@@ -2359,3 +2359,143 @@ def handle_telemetry_usage(self, response):
                 knn_vectors = field_info.get("knn_vectors_in_bytes", 0)
                 if knn_vectors > 0:
                     self.metrics_store.put_value_cluster_level("disk_usage_knn_vectors", knn_vectors, meta_data=meta, unit="byte")
+
+
+class BlobStoreStats(TelemetryDevice):
+    internal = False
+    command = "blob-store-stats"
+    human_name = "Blob Store Stats"
+    help = "Regularly samples blob store stats, only applicable to serverless Elasticsearch"
+
+    """
+    Gathers blob snapshots stats on both a cluster and node level
+    """
+
+    def __init__(self, telemetry_params, clients, metrics_store):
+        """
+        :param telemetry_params: The configuration object for telemetry_params.
+            ``blob-store-stats-sample-interval``: positive integer controlling the sampling interval.
+            Default: 1 second.
+        :param clients: A dict of clients to all clusters.
+        :param metrics_store: The configured metrics store we write to.
+        """
+        super().__init__()
+
+        self.telemetry_params = telemetry_params
+        self.clients = clients
+        self.sample_interval = telemetry_params.get("blob-store-stats-sample-interval", 1)
+        if self.sample_interval <= 0:
+            raise exceptions.SystemSetupError(
+                f"The telemetry parameter 'blob-store-stats-sample-interval' must be greater than zero " f"but was {self.sample_interval}."
```

Comment:
Nit: why the split in 2 string blocks?

Reply:
I have seen `make lint|format` (i.e. [black](https://github.com/elastic/rally/blob/a7387ae7ddc53098d640130900a6f5cf2aa5c85e/.pre-commit-config.yaml#L11-L14) automatically do that.

Reply:
Right it's unlikely to be intentional but would be nice to fix just before merging (does not need a new review)

Reply:
Just keeping you all on your toes.